### PR TITLE
add implementation stubs (fixes #285, fixes #298)

### LINKED ISSUE #285: .ml stubs for exercises
Some language tracks have a `{{ stub }}.ml` that the student can fill out.

The Ocaml track doesn't. Ideally, add stubs to all exercises.

To begin with, start with ones that use `Map` from Base, like the four exercises in #277.

Issue spawned from this message I wrote on Slack:

> I just reviewed the new version of Nucleotide Count and asked if there were any hard points, and *[the student]* said that the `(module Char)` annotations were difficult to figure out because they're not very well-documented. That was exactly the pain point that made it hard for me to convert the exercises from Core(_kernel) to Base. A solution could be to add a template .ml file with a stub in it that throws an exception. For some exercises, e.g. ones that require a data type definition repeated from the .mli, or in case of Nucleotide Count, a line like `let empty = Map.empty (module Char)`, that'd be really neat.

### LINKED ISSUE #298: Grade School's use of module parameters for Map is too hard
Both Grade School and Nucleotide Count are in the core track and rely on parameterised modules.

Nucleotide Count was migrated to use Base in #278 and a stub file was created in #295, partially to address that

    let empty = Map.empty (module Char)

is hard to figure out.

The same problem seems to apply for Grade School.

- [x] Create a stub file for Grade School
- [x] Add explaining hints to both exercises for parameterised modules

### PR DESCRIPTION
**Pull request by marionebl**
As per 
- #285 
- #298

### CODE REVIEW

**Comment by marionebl** [comment]
CI failure appears to be unrelated.

File: exercises/grade-school/grade_school.ml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+open Base
+
+module Int_map = Map.M(Int)
```
Comment (marionebl) [inline exercises/grade-school/grade_school.ml]: Adresses #298

**Comment by marionebl** [comment]
As a side note: I dug into the CI setup a bit and created this version using CircleCI:
https://github.com/marionebl/ocaml/tree/circleci-setup

**Comment by sshine** [comment]
Closing and opening PR to trigger CI. (I wonder if this is the best way.)

**Comment by sshine** [comment]
@marionebl: Could you merge master onto this feature branch?

I'm a little confused about the merge conflict; not sure if it's caused by leaving stuff out by intent, or because of breaking changes with base 0.12, or what.

**Comment by marionebl** [comment]
> @marionebl: Could you merge master onto this feature branch?

I'll get this into a mergable state as soon as the spin offs for #311 land

**Comment by marionebl** [comment]
One thing I don't like about this is the fact we do not have an automated test for the stub files as I did not find a way to build the tests without executing them.

**Comment by sshine** [comment]
@marionebl: My model track is Haskell's, and here we run all tests of all example solutions on every CI run. Come to think of it, this is somewhat wasteful. I don't really know of the particular implications of adding such a build target alias -- my practical experience with OCaml build tools are very limited -- but I like the purpose very much.

If you're interested in becoming a co-maintainer for the OCaml track, you are very welcome.

If you prefer to submit changes as you have been doing so far, you are also very welcome.

**Comment by marionebl** [comment]
> If you're interested in becoming a co-maintainer for the OCaml track, you are very welcome.
I'd be glad to!

> If you prefer to submit changes as you have been doing so far, you are also very welcome.
I'd continue submitting PRs even as maintainer so everyone stays on the same page.

**Comment by sshine** [comment]
@marionebl: Yes, that'd be ideal as long as we're more than one active maintainer, since we get the luxury of fault detection. (I don't think I can supplement your knowledge of CI and OCaml build tools, but four eyes are better than two!)

@iHiD, @kytrinyx:
- Can we make @marionebl co-maintainer of exercism/ocaml?
- And can we disable direct pushes to master? It works like this on exercism/haskell, so that all changes must come through a PR. (I'm not asking for required reviews, but rather just that we force the use of PRs, since they trigger CI.)

**Comment by iHiD** [comment]
@kytrinyx will sort these for you.

**Comment by iHiD** [comment]
@marionebl Welcome. Ping me on Slack and I'll add you to the #maintainers channel :)

**Comment by kytrinyx** [comment]
@marionebl Welcome! Thank you so much for contributing ✨

I've sent the invitation to join the exercism/ocaml team, and have turned on branch protection for the master branch. Let me know if anything is wonky, I always worry that I get things wrong when messing with repo settings.

**Pull request merged**